pkg/cloudprovider/cloudprovider.go

```diff
@@ -58,8 +58,8 @@ const (
 	maxPodsKey      = "capacity.cluster-autoscaler.kubernetes.io/maxPods"
 )
 
-func NewCloudProvider(ctx context.Context, kubeClient client.Client, machineProvider machine.Provider, machineDeploymentProvider machinedeployment.Provider) *CloudProvider {
-	return &CloudProvider{
+func NewCloudProvider(ctx context.Context, kubeClient client.Client, machineProvider machine.Provider, machineDeploymentProvider machinedeployment.Provider) CloudProvider {
```

Comment:
Hey @elmiko , I am wondering if we could keep returning the reference of the object as it looks like[1] a practice between controllers, including the cluster controller ([NewCluster](https://github.com/kubernetes-sigs/karpenter/blob/07edafd396a95d70cba4c1d1c6eff5a7bd404561/pkg/controllers/state/cluster.go#L87)) . WDYT?

[1] Checking those `New*Controller` constructors:
https://github.com/kubernetes-sigs/karpenter/blob/07edafd396a95d70cba4c1d1c6eff5a7bd404561/pkg/controllers/controllers.go#L75-L105

Is this change related to the following statement?:
> this change also includes a fix for operatorpkg to include the update label key generation for prometheus which prevents a panic in the application.


Otherwise, LGTM

Reply:
i'll need to study that deeper, when i was rebuilding the code after rebasing on the new upstream, it was complaining about the cloud provider reference.

Reply:
seems like it should probably return a pointer then we can use this `metrics.Decorate` to massage it into an interface pointer.

https://github.com/aws/karpenter-provider-aws/blob/main/cmd/controller/main.go#L42